src/SmoFacade/Server.cs

```diff
@@ -224,7 +224,29 @@ public Database Database(string dbName)
 
         policy.Execute(() => restore.SqlRestore(_server));
         restore.Devices.Remove(backupDeviceItem);
+
+        if(fileRelocation != null)
+          LogicalFileRename(databaseName, fileRelocation);
+      }

```

Comment:
I'm guessing we are doing this after each restore because it is a clean up action?
(and something we don't want to happen if the restore fails?)

Which begs the question what happens if one of iteration fails (or maybe it fails on one server).
Can we still re-run the `move()`/`restore()` function again?

Reply:
This should be done after the entire restore completes, not on each loop iteration.
if the restore fails & is retried, the db is overwritten

Reply:
Whoops, yes this was intended to be placed _after_ the loop. I have a local branch where I'm testing all the changes I've made together. I must've mistyped when copying this addition over from there!